geoana/gravity.py

```diff
@@ -0,0 +1,269 @@
+"""
+======================================================
+Gravity (:mod:`geoana.gravity`)
+======================================================
+.. currentmodule:: geoana.gravity
+
+The ``geoana.gravity`` module contains simulation classes for solving
+basic gravitational problems.
+
+Simulation Classes
+==================
+.. autosummary::
+  :toctree: generated/
+
+  PointMass
+"""
+
+import numpy as np
+
+from scipy.constants import G
+
+
+class PointMass:
+    """Class for gravitational solutions for a point mass.
+
+    The ``PointMass`` class is used to analytically compute the gravitational
+    potentials, fields, and gradients for a point mass.
+
+        Parameters
+        ----------
+        mass : float
+            Mass of the point particle (kg). Default is m = 1 kg
+        location : array_like, optional
+            Location of the point mass in 3D space. Default is (0, 0, 0)
+        """
+
+    def __init__(self, mass=1.0, location=None, **kwargs):
+
+        self.mass = mass
+        if location is None:
+            location = np.r_[0, 0, 0]
+        self.location = location
+        super().__init__(**kwargs)
+
+    @property
+    def mass(self):
+        """Mass of the point particle in kg
+
+        Returns
+        -------
+        float
+            Mass of the point particle in kg
+        """
+        return self._mass
+
+    @mass.setter
+    def mass(self, value):
+
+        try:
+            value = float(value)
+        except:
+            raise TypeError(f"mass must be a number, got {type(value)}")
+
+        self._mass = value
+
+    @property
+    def location(self):
+        """Location of the point mass
+
+        Returns
+        -------
+        (3) numpy.ndarray of float
+            Location of the point mass.  Default = np.r_[0,0,0]
+        """
+        return self._location
+
+    @location.setter
+    def location(self, vec):
+
+        try:
+            vec = np.asarray(vec, dtype=float)
+        except:
+            raise TypeError(f"location must be array_like of float, got {type(vec)}")
+
+        vec = np.squeeze(vec)
+        if vec.shape != (3,):
+            raise ValueError(
+                f"location must be array_like with shape (3,), got {vec.shape}"
+            )
+
+        self._location = vec
+
+    def gravitational_potential(self, xyz):
+        """
+        Gravitational potential due to a point mass.  See Blakely, 1996
+        equation 3.4.
+
+        .. math::
+
+            U(P) = \\gamma \\frac{m}{r}
+
+        Parameters
+        ----------
+        xyz : (..., 3) numpy.ndarray
```

Comment:
Instead of (..., 3) I think we generally use (n, 3). This would be more consistent with one of our other packages

Reply:
I disagree. This current format  makes these functions more flexible.